### ISSUE TITLE
MonetDB server crashes in `tail_type`

### ISSUE THREAD

**Issue by fuboat** [MonetDB/MonetDB#7472]
**Describe the bug**
MonetDB server crashes in `tail_type`.

**To Reproduce**
```sql
SELECT ( WITH x AS ( SELECT 1 x ) SELECT 1 FROM ( x NATURAL JOIN x ) WHERE x NOT IN ( SELECT CASE WHEN x THEN ( SELECT 1 ) END GROUP BY ( SELECT 1 ) ) );
```

**Backtrace of the Crash**
```c
#0 0x7fb1131cdb37 (tail_type+0x67)
#1 0x7fb1131dc606 (stmt_Nop+0xc26)
#2 0x7fb1131db962 (stmt_unop+0x232)
#3 0x7fb1131a1fc7 (subrel_bin+0x4aa7)
#4 0x7fb1131ab4c9 (rel2bin_project+0x1f9)
#5 0x7fb11319e1dc (subrel_bin+0xcbc)
#6 0x7fb11319e791 (subrel_bin+0x1271)
#7 0x7fb1131ab4c9 (rel2bin_project+0x1f9)
#8 0x7fb11319e1dc (subrel_bin+0xcbc)
#9 0x7fb11319d3b9 (output_rel_bin+0xb9)
#10 0x7fb1131e9a59 (backend_dumpstmt_body+0x2c9)
#11 0x7fb1131e96a2 (backend_dumpstmt+0x192)
#12 0x7fb113185184 (SQLparser+0xf04)
#13 0x7fb113183a0a (SQLengine_+0xa5a)
#14 0x7fb1131811fe (SQLengine+0x3e)
#15 0x7fb113705b2f (runScenario+0x6f)
#16 0x7fb113706cfe (MSscheduleClient+0xb6e)
#17 0x7fb11384a6c2 (doChallenge+0x132)
#18 0x7fb114667b38 (thread_starter+0x318)
#19 0x7fb113078609 (start_thread+0xd9)
#20 0x7fb112f9b353 (clone+0x43)
```

**Software versions**
 - MonetDB version number: v11.49.1 (Dec2023)
 - OS and version: Ubuntu 20.04
 - From docker image monetdb/monetdb:Dec2023 (or monetdb/dev-builds:default)

An example of steps to reproduce in docker container (**write the poc into /tmp/test.sql first**):
```bash
docker container rm monetdb_test -f
docker run -e MDB_DB_ADMIN_PASS=monetdb --name monetdb_test -itd monetdb/monetdb:Dec2023
docker exec monetdb_test bash -c 'echo -n -e "user=monetdb\npassword=monetdb\n" > /root/.monetdb'

sleep 10
cat /tmp/test.sql | timeout 5s docker exec -i monetdb_test mclient monetdb     # run test case

if docker exec monetdb_test bash -c "grep '[m]server5' /proc/*/cmdline -lR"; then
    echo "[INFO] Server normal."
else
    echo "[CRASH] Mserver5 process does NOT exist."
fi
```

It can also be reproduced on the docker image `monetdb/dev-builds:default`.

**Issue labeling**
bug

**Comment by mvdvm** [comment]
Issue appears to be in the `GROUP BY ( SELECT 1 )` part. When using `GROUP BY 1` it works.

**Comment by njnes** [comment]
Solved in dec2023
